src/event/message.rs

```diff
@@ -51,23 +51,29 @@ pub(crate) trait AttributesSerializer {
     fn serialize_attribute(&mut self, name: &str, value: MessageAttributeValue) -> Result<()>;
 }
 
-impl StructuredSerializer<Event> for Event {
-    fn set_structured_event(mut self, bytes: Vec<u8>) -> Result<Event> {
-        let new_event: Event = serde_json::from_slice(&bytes)?;
-        self.attributes = new_event.attributes;
-        self.data = new_event.data;
-        self.extensions = new_event.extensions;
-        Ok(self)
+#[derive(Debug)]
+pub(crate) struct EventStructuredSerializer {}
+
+impl StructuredSerializer<Event> for EventStructuredSerializer {
+    fn set_structured_event(self, bytes: Vec<u8>) -> Result<Event> {
+        Ok(serde_json::from_slice(&bytes)?)
     }
 }
 
-enum EventBinarySerializer {
+#[derive(Debug)]
+pub(crate) enum EventBinarySerializer {
     V10(EventBuilderV10),
     V03(EventBuilderV03)
 }
 
+impl EventBinarySerializer {
+    pub(crate) fn new() -> Self {
```

Comment:
Just confirming, is it fine to ignore v03 here?

Reply:
Yes because then with `set_spec_version` i set the right version